core/lib/Cardano/Address/Style/Icarus.hs

```diff
@@ -311,16 +313,21 @@ deriveAddressPublicKey =
 -- > "addr1vxpfffuj3zkp5g7ct6h4va89caxx9ayq2gvkyfvww48sdncxsce5t"
 --
 -- | Analyze an 'Address' to know whether it's an Icarus address or not.
--- Returns 'Nothing' if the address isn't a byron address, or return a
+-- Throws 'IcarusAddrError' if the address isn't a byron address, or return a
 -- structured JSON that gives information about an address.
 --
 -- @since 2.0.0
-inspectIcarusAddress :: Address -> Maybe Json.Value
+inspectIcarusAddress :: MonadThrow m => Address -> m Json.Value
 inspectIcarusAddress addr = do
-    payload <- CBOR.deserialiseCbor CBOR.decodeAddressPayload bytes
-    (root, attrs) <- CBOR.deserialiseCbor decodePayload payload
-    guard $ 1 `notElem` (fst <$> attrs)
-    ntwrk <- CBOR.deserialiseCbor CBOR.decodeProtocolMagicAttr payload
+    payload <- either (throwM . IcDeserialiseError) pure
+      $ CBOR.deserialiseCbor CBOR.decodeAddressPayload bytes
+    (root, attrs) <- either (throwM . IcDeserialiseError) pure
+      $ CBOR.deserialiseCbor decodePayload payload
+    _  <- if (notElem 1 . fmap fst) attrs
+             then pure ()
+             else throwM IcInvalidDerivationPath
```

Comment:
What about:

```hs
when (1 `elem` fst <$> attrs) $ 
    throwM IcInvalidDerivationPath
```

Also, I'd suggest renaming this to `UnexpectedDerivationPath` for this is the source of the problem here. We do not expected Icarus addresses to have a derivation path attributes.